packages/go_router/lib/src/delegate.dart

```diff
@@ -95,26 +94,21 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   }
 
   /// Pushes the given location onto the page stack
-  void push(RouteMatch match) {
-    if (match.route is ShellRoute) {
-      throw GoError('ShellRoutes cannot be pushed');
-    }
+  void push(RouteMatchList matches) {
```

Comment:
`replace()` calls this method with a `RouteMatch`, not a `RouteMatchList` after https://github.com/flutter/packages/pull/2747. Should `replace()` also take a `RouteMatchList`?